pkg/cvo/status.go

```diff
@@ -413,6 +429,48 @@ func updateClusterVersionStatus(cvStatus *configv1.ClusterVersionStatus, status
 	}
 }
 
+// getReasonMessageFromError returns the reason and the message from an error.
+// If the reason or message is not available, an empty string is returned.
+func getReasonMessageFromError(err error) (reason, message string) {
+	if uErr, ok := err.(*payload.UpdateError); ok {
+		reason = uErr.Reason
+	}
+	if err != nil {
+		message = err.Error()
+	}
+	return reason, message
+}
+
+// filterErrorForFailingCondition filters out update errors based on the given
+// updateEffect from a MultipleError error. If the err has the reason
+// MultipleErrors, its immediate nested errors are filtered out and the error
+// is recreated. If all nested errors are filtered out, nil is returned.
```

Comment:
NIT: Strictly speaking, it is an empty array (`[]error{}`), instead of `nil`.
I do not have preference between updating the code or comment.
I am happy as long as they are consistent.

Reply:
`func filterOutUpdateErrors(errs []error, updateEffect payload.UpdateEffectType) []error {` returns an empty array (`[]error{}`).

`func filterErrorForFailingCondition(err error, updateEffect payload.UpdateEffectType) error {` for which the comment is made returns `nil`. 
```go
...
filtered = filterOutUpdateErrors(filtered, updateEffect) // filtered is []error{}
return newMultipleError(filtered)                        // newMultipleError returns nil for an empty array
...
```